src/main/java/org/jboss/ejb/client/remoting/ChannelAssociation.java

```diff
@@ -161,7 +161,16 @@ EJBReceiverContext getEjbReceiverContext() {
      * @return
      */
     short getNextInvocationId() {
-        return (short) nextInvocationId.getAndIncrement();
+        for (;;) {
+            short nextId = (short) nextInvocationId.getAndIncrement();
+            if (this.waitingMethodInvocations.containsKey(nextId)) {
+                continue;
+            }
+            if (this.waitingFutureResults.containsKey(nextId)) {
+                continue;
+            }
+            return nextId;
+        }
```

Comment:
There is a short time span between generating an ID and the ID beeing inserted into one of those maps (waitingMethodInvocations, waitingFutureResults), but conflict would only happen if there was 65536 other IDs generated in that timespan, which is unlikely.